src/camera/Camera.cpp

```diff
@@ -115,23 +136,26 @@ bool Camera::openFromConfigFile(std::string filename) {
 
 void Camera::captureLoop() {
 	loguru::set_thread_name(_name.c_str());
-	cv::Size image_size(640, 480);
-	if (!_intrinsic_params.empty()) {
-		image_size = _intrinsic_params.getImageSize();
-	}
-	_capture_lock->lock();
-	_capture->set(cv::CAP_PROP_FOURCC, cv::VideoWriter::fourcc('M', 'J', 'P', 'G'));
-	_capture->set(cv::CAP_PROP_FRAME_WIDTH, image_size.width);
-	_capture->set(cv::CAP_PROP_FRAME_HEIGHT, image_size.height);
-	_capture_lock->unlock();
 	cv::Mat frame;
 	while (*_running) {
 		_capture_lock->lock();
 		bool success = _capture->read(frame);
 		_capture_lock->unlock();
 		if (success && !frame.empty()) {
+			// Aruco detection here:
+			cv::Ptr<cv::aruco::Dictionary> dictionary =
+				cv::aruco::getPredefinedDictionary(cv::aruco::DICT_4X4_100);
+			std::vector<std::vector<cv::Point2f>> markerCorners;
+			cv::Mat frameCopy;
+			std::vector<int> markerIds;
+			cv::aruco::detectMarkers(frame, dictionary, markerCorners, markerIds);
+			frame.copyTo(frameCopy);
+			if (!markerIds.empty()) {
+				cv::aruco::drawDetectedMarkers(frameCopy, markerCorners, markerIds);
+			}
```

Comment:
This code does aruco tag detection (not localization!). We default to doing it on every frame of every camera, but maybe that should be configurable? Or maybe detection shouldn't be in the same file as camera reading. Maybe overhaul the entirety of the `AR` codebase and do detection/localization in there. In any case, this is a code sample that works so use as you see fit.